lib/evaluation/handlers/sameSubjectEvaluationHandler.ts

```diff
@@ -184,15 +166,13 @@ export class SameSubjectEvaluationHandler extends AbstractEvaluationHandler {
   ) {
     const entry = this.getValue(fieldIdzInputDescriptors, searchableFieldIds);
 
-    searchableFieldIds.forEach((searchableFieldId) => entry.mappedFieldIds.add(searchableFieldId));
```

Comment:
Why not have an if around both foreach lines instead of allowing an undefined entry in both foreaches. They will do nothing if entry is undefined